bgcval2/analysis_timeseries.py

```diff
@@ -139,6 +163,180 @@ def build_list_of_suite_keys(suites, debug=True):
     return analysis_keys
 
 
+def load_function(functionname):
+    """
+    Using the named function in the key yaml, load that function and return it. 
+    """
+    # load functions:
+    if functionname in std_functions.keys():
+        print( "Standard Function Found:", functionname)
+        func = std_functions[functionname]
+
+    if functionname.find(':') > -1:
+        [functionFileName,functionname] = functionname.split(':')
+        lst = functionFileName.replace('.py','').replace('/', '.').split('.')
+        modulename =  '.'.join(lst)
+
+        print("parseFunction:\tAttempting to load the function:",functionname, "from the:",modulename)
+        mod = __import__(modulename, fromlist=[functionname,])
+        func = getattr(mod, functionname)
+    return func     
+
+
+def load_function_kwargs(dictionary, m_or_d):
+    """
+    Loads key word arguments from yaml file dictionairy.
+    """
+    data_kwargs = {}
+    #, data_convert_kwargss = [], []
+    #####
+    # Looking for kwargs to pass to convert:
+    for key, value in dictionary.items():
+        searchFor =  m_or_d+'_convert_'
+        findstr = key.find(searchFor)
+        if findstr==-1: continue
+        kwargkey = key[len(searchFor):]
+        data_kwargs[kwargkey] = value
+    return data_kwargs 
+
+
+def findReplaceFlag(filepath, flag, new_value):
+    """
+    Looking for $FLAG in the filepath.
+    If found, we replace $FLAG with new_value
+    """
+    lookingFor = '$'+flag.upper()
+    if filepath.find(lookingFor) ==-1: 
```

Comment:
```suggestion
    if not filepath.find(lookingFor): 
```

Reply:
but if it finds `lookingFor` at the start of the string, it will return zero. If it doesn't find it, it will return `-1`.

Reply:
right! That's a numeral return, correct, I've not used the `find` functionality in ages - that;s why it's better to go with the string search in string `if not "lookingFor" in filepath:` - I think it's also more performant